pkg/cli/config.go

```diff
@@ -94,3 +96,67 @@ func buildConfig(sources []io.Reader) (*Config, error) {
 	}
 	return c, nil
 }
+
+type AcloudConfig struct {
+	Zone        string
+	MachineType string
+}
+
+func ImportAcloudConfig(src, dst string) error {
+	b, err := os.ReadFile(src)
+	if err != nil {
+		return fmt.Errorf("error reading %q: %w", src, err)
+	}
+	ac, err := extractAcloudConfig(string(b))
+	if err != nil {
+		return fmt.Errorf("failed extracing acloud config: %w", err)
+	}
+	c := map[string]any{
+		"Zone": ac.Zone,
+		"Host": map[string]any{
+			"GCP": map[string]any{
+				"MachineType": ac.MachineType,
+			},
+		},
+	}
+	dstDir := filepath.Dir(dst)
+	if err := os.MkdirAll(filepath.Dir(dst), 0750); err != nil {
+		return fmt.Errorf("failed creating dir %q: %w", dstDir, err)
+	}
+	f, err := os.Create(dst)
+	if err != nil {
+		return fmt.Errorf("failed creating file %q: %w", dst, err)
+	}
+	defer f.Close()
+	encoder := toml.NewEncoder(f).Indentation("")
+	if err = encoder.Encode(c); err != nil {
+		return fmt.Errorf("failed encoding: %w", err)
+	}
+	return nil
+}
+
+// Extract relevant acloud config values.
+// Acloud config is written in Proto Text Format https://protobuf.dev/reference/protobuf/textformat-spec/
+func extractAcloudConfig(input string) (AcloudConfig, error) {
+	extract := func(field, input string) (string, error) {
+		re := regexp.MustCompile(`(^|\n)` + field + `:\s+\"([^"]+)\"`)
```

Comment:
This regex will miss lines like these:
```
field:"value"
   field: "value"
field : "value"
filed: "val\"ue"
```
A better regex would be `[\n^]\s*` + field + `\s*:\s*\"(.*)\"\s*[\n$]`, note that the value will be in `matches[1]`, not `matches[2]`.

Reply:
good catch, added the whitespaces and used a stricter group matching due the nature of these values.